core/src/main/scala/chisel3/experimental/HasChiselTarget.scala

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.experimental
+import firrtl.annotations.{CompleteTarget, SingleTargetAnnotation}
+
+/** An Annotation that records the original target annotate from Chisel. */
+trait HasChiselTarget[T <: CompleteTarget] { x: SingleTargetAnnotation[T] =>
```

Comment:
Why did you make this trait, when it is only used by a single class?
Seems like your code would be simpler without making this a separate trait.